util/metric/metric.go

```diff
@@ -205,15 +232,29 @@ func (c *Counter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Counter.Count())
 }
 
+// Desc implements prometheus.Metric.
+func (c *Counter) Desc() *prometheus.Desc {
+	return c.desc
+}
+
+// Write implements prometheus.Metric.
+func (c *Counter) Write(met *prometheus_model.Metric) error {
+	met.Gauge = &prometheus_model.Gauge{Value: proto.Float64(float64(c.Count()))}
```

Comment:
Oops, yeah, good point indeed. I was only referring to the buckets at a given point in time (cumulative in the sense the higher latency buckets contain the lower latencies, too). Cumulative over time is a different question. It's part of the Prometheus data model that all buckets are just counters that get incremented forever (modulo counter resets). So only when evaluating expressions on the Prometheus server, you have to decide which sliding window you want to take (which is obviously a great advantage oven picking the sliding window during collection already).

To reconcile both models might be some effort...